ibm_mq/datadog_checks/ibm_mq/collectors/queue_metric_collector.py

```diff
@@ -141,6 +151,82 @@ def _discover_queues(self, queue_manager, mq_pattern_filter):
+        # type: (pymqi.QueueManager, str) -> List[str]
+        queues = []
+
+        for queue_type in SUPPORTED_QUEUE_TYPES:
+            args = {pymqi.CMQC.MQCA_Q_NAME: pymqi.ensure_bytes(mq_pattern_filter), pymqi.CMQC.MQIA_Q_TYPE: queue_type}
+            pcf = None
+            try:
+                pcf = pymqi.PCFExecute(
+                    queue_manager, response_wait_interval=self.config.timeout, convert=self.config.convert_endianness
+                )
+                # Use MQCMD_INQUIRE_Q_NAMES to get only the queue names rather than the full queue info
+                response = pcf.MQCMD_INQUIRE_Q_NAMES(args)
+                queue_names = response[0].get(pymqi.CMQCFC.MQCACF_Q_NAMES, []) if response else []
+                for queue in queue_names:
+                    queue_name = to_string(queue).strip()
+                    if not queue_name:
+                        self.log.debug('Discovered queue with empty name, skipping.')
+                        continue
+                    # For each queue name inquire the queue info
+                    inquire_args = {
+                        pymqi.CMQC.MQCA_Q_NAME: pymqi.ensure_bytes(queue_name),
+                        pymqi.CMQC.MQIA_Q_TYPE: queue_type,
+                    }
+                    try:
+                        queue_info_response = pcf.MQCMD_INQUIRE_Q(inquire_args)
+                        if queue_info_response:
+                            self.log.debug("Discovered queue: %s", queue_name)
+                            queues.append(queue_name)
+                    except pymqi.MQMIError as e:
+                        # Don't warn if no messages, see:
+                        # https://github.com/dsuch/pymqi/blob/v1.12.0/docs/examples.rst#how-to-wait-for-multiple-messages
+                        if e.comp == pymqi.CMQC.MQCC_FAILED and e.reason == pymqi.CMQC.MQRC_NO_MSG_AVAILABLE:
+                            self.log.debug("No queue info available for queue %s", queue_name)
+                        elif e.comp == pymqi.CMQC.MQCC_FAILED and e.reason == pymqi.CMQC.MQRC_UNKNOWN_OBJECT_NAME:
+                            self.log.debug("No matching queue of type %d for queue %s", queue_type, queue_name)
+                        else:
+                            self.log.debug("Error inquiring queue %s: %s", queue_name, e)
+                            self._submit_discovery_error_metric(e, [f"queue:{queue_name}"])
+                            continue
+                self.log.debug("%s queues discovered", str(len(queues)))
+            except pymqi.MQMIError as e:
+                self.log.debug("Error inquiring queue names for pattern %s: %s", mq_pattern_filter, e)
+                self._submit_discovery_error_metric(e, [f"queue_pattern:{mq_pattern_filter}"])
+                continue
+            except Exception as e:
+                self.log.debug("Error retrieving queue info for %s: %s", mq_pattern_filter, e)
+                continue
```

Comment:
why do we `continue` here? in what cases should we disconnect?

Reply:
I don't think it's needed. I think I forgot to refactor it out. I initially had it inside the other _discover_queues function and in there there was a continue block here that I didn't really understand and I left it in just incase it necessary. But, given the way the code is structured now, there's an implicit continue anyways and it probably isn't needed.

https://github.com/DataDog/integrations-core/blob/master/ibm_mq/datadog_checks/ibm_mq/collectors/queue_metric_collector.py#L130